src/PhpSpreadsheet/ReferenceHelper.php

```diff
@@ -608,7 +608,7 @@ public function insertNewBefore($pBefore, $pNumCols, $pNumRows, Worksheet $pShee
         // Update workbook: define names
         if (count($pSheet->getParent()->getDefinedNames()) > 0) {
             foreach ($pSheet->getParent()->getDefinedNames() as $definedName) {
-                if ($definedName->getWorksheet()->getHashCode() === $pSheet->getHashCode()) {
+                if ($definedName->getWorksheet() !== null && $definedName->getWorksheet()->getHashCode() === $pSheet->getHashCode()) {
```

Comment:
@MarkBaker does this make sense? 
I don't think updateCellReference should be called on a null worksheet

Reply:
Makes perfect sense. thanks for the quick update